### ISSUE TITLE
Can no longer authenticate in 2024.10.1

### ISSUE THREAD

**Issue by Tuxrug** [natekspencer/hacs-vivint#148]
### The problem

After updating from 2024.7.0 to 2024.10.1, I get an error when starting the integration.

```
Logger: homeassistant.config_entries
Source: config_entries.py:594
First occurred: 6:13:59 PM (1 occurrences)
Last logged: 6:13:59 PM

Error setting up entry homeassistant@tuxrug.net for vivint
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 594, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/vivint/__init__.py", line 64, in async_setup_entry
    await hub.login(load_devices=True, subscribe_for_realtime_updates=True)
  File "/config/custom_components/vivint/hub.py", line 96, in login
    self.account = Account(
                   ^^^^^^^^
TypeError: Account.__init__() got an unexpected keyword argument 'refresh_token'

```
I tried removing the integration and re-adding it but still got `TypeError: Account.__init__() got an unexpected keyword argument 'refresh_token'` when trying to log in at the username/password stage. This occurred both with and without MFA enabled for the account it is using. Reverting to 2024.7.0 allows it to log in again with a new emailed verification code.

### What version of Home Assistant Core are you running?

core-2024.10.1

**Comment by natekspencer** [comment]
What version of home assistant core are you running? It looks like it maybe didn't download the latest vivintpy library appropriately based on the error and I'm wondering if there might be a conflict. And can you try again, restart home assistant and upload the logs to see if there are any issues prior to that?

**Comment by Tuxrug** [comment]
[home-assistant_2024-10-11T01-00-51.886Z.log](https://github.com/user-attachments/files/17336022/home-assistant_2024-10-11T01-00-51.886Z.log)
I'm using the core 2024.10.1 docker image in podman.

**Comment by Tuxrug** [comment]
This still occurs in HA core-2024.10.2. I found the vivintpy folder where __init__.py listed version 2024.1.1. I renamed the vivintpy and vivintpy-2024.1.1.dist-info directories, re-updated through HACS, and restarted HA which re-created the vivintpy and dist-info folders, still referencing 2024.1.1 and I still receive `TypeError: Account.__init__() got an unexpected keyword argument 'refresh_token'` in the logs.

I removed the vivintpy folder and downgraded this integration to 2024.10.1b0 and restarted, and saw it create download vivintpy 2024.1.0 and then updated to b1 which downloaded vivintpy 2024.1.1. The issue still continues.

**Comment by natekspencer** [comment]
I've been unable to reproduce this unfortunately. The error seems to indicate that refresh token is being passed in but not expected, so it seems like the custom component code is correct (trying to use a refresh token), but the vivintpy account seems to not be expecting it, which makes me think it's still somehow conflicting with something else.

**Comment by Tuxrug** [comment]
Is there anything I can do on my end to help narrow it down further?

**Comment by natekspencer** [comment]
Are you able to bash into your docker container and run `pip list | grep vivint`?

**Comment by Tuxrug** [comment]
That command returns nothing, which surprises me because I'm currently running 2024.7.0 without any issues. It also returns nothing after redownloading 2024.10.1. `pip list` is returning things, just nothing matching the string "vivint", and the vivintpy folder is recreated in deps on restart if I rename it.

I tried `pip install vivintpy` and now `pip list` returns vivintpy 2024.2.0. However restarting the container causes it to no longer be listed, so it seems like I'm maybe not persisting something in the container that I should be and/or something is keeping vivintpy from installing correctly. Same thing happens if I do `pip install vivintpy --target /config/deps` but I do receive additional warnings/error output mostly around version conflicts doing that. 

```
Installing collected packages: urllib3, pyjwt, pycryptodomex, protobuf, propcache, multidict, idna, grpcio, frozenlist, charset-normalizer, certifi, cbor2, attrs, aiohappyeyeballs, yarl, requests, aiosignal, aiohttp, pubnub, vivintpy
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
homeassistant 2024.10.2 requires aiohttp==3.10.8, but you have aiohttp 3.10.10 which is incompatible.
homeassistant 2024.10.2 requires attrs==23.2.0, but you have attrs 24.2.0 which is incompatible.
homeassistant 2024.10.2 requires urllib3<2,>=1.26.5, but you have urllib3 2.2.3 which is incompatible.
homeassistant 2024.10.2 requires yarl==1.13.1, but you have yarl 1.15.2 which is incompatible.
freebox-api 1.1.0 requires urllib3<2.0.0,>=1.26.6, but you have urllib3 2.2.3 which is incompatible.
meteofrance-api 1.3.0 requires urllib3<2.0.0,>=1.26.18, but you have urllib3 2.2.3 which is incompatible.
weheat 2024.9.23 requires urllib3<2.1.0,>=1.25.3, but you have urllib3 2.2.3 which is incompatible.
zhong-hong-hvac 1.0.12 requires attrs<24.0.0,>=23.1.0, but you have attrs 24.2.0 which is incompatible.
Successfully installed aiohappyeyeballs-2.4.3 aiohttp-3.10.10 aiosignal-1.3.1 attrs-24.2.0 cbor2-5.6.5 certifi-2024.8.30 charset-normalizer-3.4.0 frozenlist-1.4.1 grpcio-1.66.2 idna-3.10 multidict-6.1.0 propcache-0.2.0 protobuf-4.25.5 pubnub-8.0.0 pycryptodomex-3.21.0 pyjwt-2.9.0 requests-2.32.3 urllib3-2.2.3 vivintpy-2024.2.0 yarl-1.15.2
WARNING: Target directory /config/deps/urllib3 already exists. Specify --upgrade to force replacement.
WARNING: Target directory /config/deps/jwt already exists. Specify --upgrade to force replacement.
WARNING: Target directory /config/deps/PyJWT-2.9.0.dist-info already exists. Specify --upgrade to force replacement.
WARNING: Target directory /config/deps/pycryptodomex-3.21.0.dist-info already exists. Specify --upgrade to force replacement.
WARNING: Target directory /config/deps/Cryptodome already exists. Specify --upgrade to force replacement.
WARNING: Target directory /config/deps/google already exists. Specify --upgrade to force replacement.
WARNING: Target directory /config/deps/propcache-0.2.0.dist-info already exists. Specify --upgrade to force replacement.
WARNING: Target directory /config/deps/propcache already exists. Specify --upgrade to force replacement.
WARNING: Target directory /config/deps/multidict-6.1.0.dist-info already exists. Specify --upgrade to force replacement.
WARNING: Target directory /config/deps/multidict already exists. Specify --upgrade to force replacement.
WARNING: Target directory /config/deps/idna already exists. Specify --upgrade to force replacement.
WARNING: Target directory /config/deps/idna-3.10.dist-info already exists. Specify --upgrade to force replacement.
WARNING: Target directory /config/deps/grpc already exists. Specify --upgrade to force replacement.
WARNING: Target directory /config/deps/grpcio.libs already exists. Specify --upgrade to force replacement.
WARNING: Target directory /config/deps/frozenlist-1.4.1.dist-info already exists. Specify --upgrade to force replacement.
WARNING: Target directory /config/deps/frozenlist already exists. Specify --upgrade to force replacement.
WARNING: Target directory /config/deps/charset_normalizer already exists. Specify --upgrade to force replacement.
WARNING: Target directory /config/deps/certifi already exists. Specify --upgrade to force replacement.
WARNING: Target directory /config/deps/certifi-2024.8.30.dist-info already exists. Specify --upgrade to force replacement.
WARNING: Target directory /config/deps/_cbor2.cpython-312-x86_64-linux-musl.so already exists. Specify --upgrade to force replacement.
WARNING: Target directory /config/deps/cbor2-5.6.5.dist-info already exists. Specify --upgrade to force replacement.
WARNING: Target directory /config/deps/cbor2 already exists. Specify --upgrade to force replacement.
WARNING: Target directory /config/deps/attr already exists. Specify --upgrade to force replacement.
WARNING: Target directory /config/deps/attrs already exists. Specify --upgrade to force replacement.
WARNING: Target directory /config/deps/aiohappyeyeballs already exists. Specify --upgrade to force replacement.
WARNING: Target directory /config/deps/aiohappyeyeballs-2.4.3.dist-info already exists. Specify --upgrade to force replacement.
WARNING: Target directory /config/deps/yarl already exists. Specify --upgrade to force replacement.
WARNING: Target directory /config/deps/requests already exists. Specify --upgrade to force replacement.
WARNING: Target directory /config/deps/requests-2.32.3.dist-info already exists. Specify --upgrade to force replacement.
WARNING: Target directory /config/deps/aiosignal already exists. Specify --upgrade to force replacement.
WARNING: Target directory /config/deps/aiosignal-1.3.1.dist-info already exists. Specify --upgrade to force replacement.
WARNING: Target directory /config/deps/aiohttp already exists. Specify --upgrade to force replacement.
WARNING: Target directory /config/deps/pubnub already exists. Specify --upgrade to force replacement.
WARNING: Target directory /config/deps/pubnub-8.0.0.dist-info already exists. Specify --upgrade to force replacement.
WARNING: Target directory /config/deps/vivintpy already exists. Specify --upgrade to force replacement.
WARNING: Target directory /config/deps/bin already exists. Specify --upgrade to force replacement.
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager, possibly rendering your system unusable.It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv. Use the --root-user-action option if you know what you are doing and want to suppress this warning.
```

I'm running HA as privileged in podman and mapping the entire /config directory to my local filesystem (`-v /srv/homeassistant:/config`)

**Comment by smgilliam** [comment]
I am also getting this after the last update.

i have version 2024.10.1 running on a pi

2024-10-16 18:36:36.293 WARNING (MainThread) [homeassistant.bootstrap] Waiting on integrations to complete setup: {('vivint', '6b2bf93645ebc1ec51a44dc2daad94ea'): 90605.152042556}
2024-10-16 18:36:47.534 ERROR (MainThread) [custom_components.vivint.hub] Unable to connect to the Vivint API
2024-10-16 18:36:50.242 ERROR (SyncWorker_15) [homeassistant] Error doing job: Unclosed client session (None)
2024-10-16 18:36:50.247 ERROR (SyncWorker_15) [homeassistant] Error doing job: Unclosed connector (None)
2024-10-16 18:37:29.737 ERROR (MainThread) [custom_components.vivint.hub] Unable to connect to the Vivint API
2024-10-16 18:38:49.347 ERROR (SyncWorker_12) [homeassistant] Error doing job: Unclosed client session (None)
2024-10-16 18:38:58.329 ERROR (MainThread) [custom_components.vivint.hub] Unable to connect to the Vivint API
2024-10-16 18:39:18.637 DEBUG (MainThread) [vivintpy.account] Connecting to VivintSky

**Comment by natekspencer** [comment]
> I am also getting this after the last update.
> 
> i have version 2024.10.1 running on a pi
> 
> 2024-10-16 18:36:36.293 WARNING (MainThread) [homeassistant.bootstrap] Waiting on integrations to complete setup: {('vivint', '6b2bf93645ebc1ec51a44dc2daad94ea'): 90605.152042556} 2024-10-16 18:36:47.534 ERROR (MainThread) [custom_components.vivint.hub] Unable to connect to the Vivint API 2024-10-16 18:36:50.242 ERROR (SyncWorker_15) [homeassistant] Error doing job: Unclosed client session (None) 2024-10-16 18:36:50.247 ERROR (SyncWorker_15) [homeassistant] Error doing job: Unclosed connector (None) 2024-10-16 18:37:29.737 ERROR (MainThread) [custom_components.vivint.hub] Unable to connect to the Vivint API 2024-10-16 18:38:49.347 ERROR (SyncWorker_12) [homeassistant] Error doing job: Unclosed client session (None) 2024-10-16 18:38:58.329 ERROR (MainThread) [custom_components.vivint.hub] Unable to connect to the Vivint API 2024-10-16 18:39:18.637 DEBUG (MainThread) [vivintpy.account] Connecting to VivintSky
> 
> EDIT: After rebooting my Vivint panel, it works

Yours was related to Vivint service interruptions I've been seeing today.

**Comment by Tuxrug** [comment]
I tried again after updating Home Assistant to 2024.11.0 and updating hacs-vivint to 2024.11.1, but the results were the same. I then tried backing up and removing my deps folder and got this error I don't recall getting previously:

```
Logger: homeassistant.setup
Source: setup.py:334
First occurred: 4:28:43 PM (1 occurrences)
Last logged: 4:28:43 PM

Setup failed for custom integration 'vivint': Unable to import component: No module named 'vivintpy'
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/setup.py", line 334, in _async_setup_component
    component = await integration.async_get_component()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1031, in async_get_component
    self._component_future.result()
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1011, in async_get_component
    comp = await self.hass.async_add_import_executor_job(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1071, in _get_component
    ComponentProtocol, importlib.import_module(self.pkg_path)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/util/loop.py", line 200, in protected_loop_func
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 995, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/config/custom_components/vivint/__init__.py", line 8, in <module>
    from vivintpy.devices import VivintDevice
ModuleNotFoundError: No module named 'vivintpy'
```

This same issue occurs after removing all custom_components and deps again leaving only hacs-vivint, and after re-installing and reconfiguring hacs. After doing this,, `pip list --path /config/deps` does list vivintpy
```
Package            Version
------------------ ---------
aiohappyeyeballs   2.4.3
aiohttp            3.10.10
aiosignal          1.3.1
attrs              24.2.0
cbor2              5.6.5
certifi            2024.8.30
charset-normalizer 3.2.0
frozenlist         1.5.0
grpcio             1.66.2
idna               3.10
multidict          6.1.0
propcache          0.2.0
protobuf           5.28.3
pubnub             8.0.0
pycryptodomex      3.21.0
PyJWT              2.9.0
requests           2.32.3
urllib3            1.26.20
vivintpy           2024.2.1
yarl               1.17.1
```
However, `pip list | grep vivint` still returns nothing, Based on this discovery and the error it sounds like Home Assistant is downloading and installing the dependencies to /config/deps but not looking for them there.

Downgrading gave me errors below (it looks like protobuf was bumped in the new HA image), so I restored a backup and am back on 2024.10.4 with hacs-vivint version 2024.7.0
```
Logger: homeassistant.util.package
Source: util/package.py:152
First occurred: 3:36:55 PM (3 occurrences)
Last logged: 3:37:01 PM

Unable to install package vivintpy==2024.1.1: × No solution found when resolving dependencies: ╰─▶ Because vivintpy==2024.1.1 depends on protobuf>=4.23.1,<5.0.0 and protobuf==5.28.3, we can conclude that vivintpy==2024.1.1 cannot be used. And because you require vivintpy==2024.1.1, we can conclude that your requirements are unsatisfiable.
```

**Comment by natekspencer** [comment]
You're configuration confounds me and I have no solution for you unfortunately. It looks like maybe there could be an issue with podman, HACS 2.0 and custom components though: https://community.home-assistant.io/t/requirements-of-custom-integrations-not-available/788289

**Comment by Tuxrug** [comment]
Thanks, I hadn't found that despite searching the ModuleNotFound error. Considering I've been unable to get Bitwarden hosting working with podman I was wondering if that was an issue too and even tried disabling SELinux on my host to no avail, but it befuddles me why it's able to use the old version of vivintpy even if it's deleted and redownload on restart (as long as I stay on 2024.10.x). I have a spare device I could throw HAOS on and just update my existing nginx to point to that if push comes to shove; I'll start by loading a backup into a VM on my laptop to test.

**Comment by Tuxrug** [comment]
> You're configuration confounds me and I have no solution for you unfortunately. It looks like maybe there could be an issue with podman, HACS 2.0 and custom components though: https://community.home-assistant.io/t/requirements-of-custom-integrations-not-available/788289

I haven't gotten a chance yet to test with something other than podman, however the workaround maleadt used (adding the following code at the top of the __init__.py for the affected custom_components) is working perfectly for me so far:
```
import sys
sys.path.insert(0, "/config/deps")
```

I'm not well-versed in python or HA's internal workings to know if there would be any unintended consequences of doing this, but immediately after adding this code to my copy of hacs-vivint in __init__.py and restarting, it prompted for me to update the authentication and began working. 

The fact that you're unable to replicate the issue and nobody else is corroborating my issue tells me that under normal circumstances, this is already in the path. Since I don't work with python (yet), I worry that this being in the release could break the path for people who already have a correct one set. Would testing if the path already contains `/config/deps` and only inserting it if not found be a good solution?

**Comment by jsternadel** [comment]
I just got an issue with conflicting dependencies for vivintpy when upgrading to 2024.11.0 but unfortunately I rolled back before grabbing the details from the logs.  

I just wanted to check if anyone else was having the same type of issue before I break my install again so I can get the error logs.

**Comment by natekspencer** [comment]
> Would testing if the path already contains `/config/deps` and only inserting it if not found be a good solution?

It works for you, but it doesn't make sense to add to the custom component repo permanently since this seems unique to your configuration/setup. For example, if a podman2 setup comes along or some other specialized container that changes to something like `/config/hacs/custom` etc, adding each of these potential path mappings becomes untenable. The proper change is somewhere upstream.

I'm going to close this issue as it appears to be unrelated to the custom component itself. If you find otherwise (or an official stance from Home Assistant and/or HACS, please feel free to open a new issue and we can go from there.

**Comment by natekspencer** [comment]
> I just got an issue with conflicting dependencies for vivintpy when upgrading to 2024.11.0 but unfortunately I rolled back before grabbing the details from the logs.
> 
> I just wanted to check if anyone else was having the same type of issue before I break my install again so I can get the error logs.

@jsternadel You need to update your version of HACS-Vivint (released last week) to resolve the vivintpy dependency issue.

**Comment by Tuxrug** [comment]
Apologies if this re-opens the issue, please re-close if it does - I just wanted to let anyone else who may search for this that this appears to be an open issue in HA itself now and there's a more manageable workaround than the one I posted above:

https://github.com/home-assistant/core/issues/133537

The workaround in this post (adding a volume pointing to `/.dockerenv` inside the container) is working marvelously.
